ibis/backends/base/sql/alchemy/registry.py

```diff
@@ -81,24 +81,24 @@ def get_sqla_table(ctx, table):
     return sa_table
 
 
-def get_col_or_deferred_col(sa_table, colname):
-    """Get a `Column`, or create a "deferred" column.
+def get_col(sa_table, op: ops.TableColumn):
+    """Get a `Column`."""
+    cols = sa_table.exported_columns
+    colname = op.name
 
-    This is to handle the case when selecting a column from a join, which
-    happens when a join expression is cached during join traversal
-
-    We'd like to avoid generating a subquery just for selection but in
-    sqlalchemy the Join object is not selectable. However, at this point
-    know that the column can be referred to unambiguously
-
-    Later the expression is assembled into
-    `sa.select([sa.column(colname)]).select_from(table_set)` (roughly)
-    where `table_set` is `sa_table` above.
-    """
     try:
-        return sa_table.exported_columns[colname]
+        return cols[colname]
     except KeyError:
-        return sa.column(colname)
+        # XXX: cols is a sqlalchemy column collection which contains column
+        # names that are secretly prefixed by their containing table
+        #
+        # sqlalchemy doesn't let you select by the *un*prefixed column name
+        # despite the uniqueness of `colname`
+        #
+        # however, in ibis we have already deduplicated column names so we can
+        # refer to the name by position
```

Comment:
```suggestion
        # cols is a sqlalchemy column collection which contains column
        # names that are secretly prefixed by their containing table
        #
        # sqlalchemy doesn't let you select by the *un*prefixed column name
        # despite the uniqueness of `colname`
        #
        # however, in ibis we have already deduplicated column names so we can
        # refer to the name by position
```

The micro-ist of nits, but to me `# XXX` reads like something that needs fixing before merge, but that is not a blocker at all.

Reply:
Fixed!